sherpa/models/model.py

```diff
@@ -46,12 +46,12 @@ def cache_model(cls, pars, xlo, *args, **kwargs):
         digest = ''
         if use_caching:
 
-            data = [ numpy.array(pars).tostring(), str(kwargs.get('integrate',0)), 
+            data = [ numpy.array(pars).tostring(), kwargs.get('integrate', b'0'),
```

Comment:
the keyword is actually set in several tests (implicitly by the model classes themselves), but the potential bug would only be triggered by models that enable caching *and* have `integrate=True`, which is never the case in the tests. I am not even sure what the expected behavior in should be in this corner case.

Also, this looks like a potential bug in the current code as well, right @DougBurke?